apps/04TowersOfHanoi.js

```diff
@@ -13,30 +13,108 @@ var stacks = {
   c: []
 };
 
+//reset board when game is won
+function reset() {
+  stacks = {
+    a: [4, 3, 2, 1],
+    b: [],
+    c: []
+  };
+}
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack, endStack) {
   // Your code here
+  var move = stacks[startStack].pop();
+  stacks[endStack].push(move);
 
 }
 
-function isLegal() {
+function isLegal(startStack, endStack) {
   // Your code here
 
+  //check for endStack value
+  var check = stacks[endStack];
+  var check2 = check[check.length-1];
+  //added console.log to see what value was being produced
+  //console.log("isLegal end ", check2);
+  
+  //create zeroCheck value in order to clear an undefined value production later on
+  var zeroCheck = check.length;
+
+  //check for startStack value
+  var cheek = stacks[startStack];
+  var cheek2 = cheek[cheek.length-1];
+  //value check again, commented out when program was finished
+  //console.log('start ', cheek2);
+  
+  //check to see if end spot is bigger that start stop
+  //test for zero, was getting an undefined value here
+  //created zeroCheck to specifically check for zero and not undefined. 
+  if (zeroCheck === 0){
```

Comment:
You could also pass in `!check2` instead of `zeroCheck` for this if statement on line 59. 

If `check2` is undefined, you know that endStack is empty.